tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_client_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    # Communication Timeline
+    # Client [S2N]                                 | Server [OpenSSL]
+    # Handshake                                    | Padded handshake
+    # Send data                                    | Receive data
+    # Receive padded data                          | Send padded data on send_marker client's sent data
+    # Close on close_marker server's padded data   | Close
+    port = next(available_ports)
+
+    client_random_bytes = data_bytes(PAYLOAD_SIZE)
+    server_random_bytes = data_bytes(PAYLOAD_SIZE)
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        cert=certificate.cert,
+        key=certificate.key,
+        insecure=True,
+        protocol=protocol,
+        data_to_send=server_random_bytes,
+        extra_flags=[
+            '-record_padding', padding_size]
+    )
+
+    client_options = copy.copy(server_options)
+    client_options.mode = Provider.ClientMode
+    client_options.extra_flags = None
+    client_options.data_to_send = client_random_bytes
+
+    # s2nc will wait until it has received the server's response before closing
+    s2nc = managed_process(S2N, client_options, timeout=5,
+                           close_marker=str(server_random_bytes)[2:-1])
+
+    # openssl will send it's response after it has received s2nc's record
+    openssl = managed_process(provider, server_options,
+                              timeout=5, send_marker=str(client_random_bytes)[2:-1])
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+    for client_results in s2nc.get_results():
+        client_results.assert_success()
+        # assert that the client has received server's application payload
+        assert server_random_bytes in client_results.stdout
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in client_results.stdout
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in client_results.stdout
+
+    for server_results in openssl.get_results():
+        server_results.assert_success()
+
+        # find all occurrences of openssl writing to client
+        all_occurrences = re.findall(
+            OPENSSL_WRITE_TO_REGEX, str(server_results.stdout))
```

Comment:
You need to assert the number of occurrences, or this test passes while matching nothing. You don't need to be exact, but >0 at a minimum, or since there's really no variability in your handshake, the number of handshake messages.